### DISCUSSION TITLE
Two-Line Prompt Theme Issue when Using Oh My Posh

### DISCUSSION THREAD

**Discussion by SetsuikiHyoryu** [nushell/nushell#14021] [Q&A] [answered]
I've encountered an issue with the Two-Line Prompt Theme when using Oh My Posh on Windows. The problem occurs when I execute the following steps:

1. Press `Win + R` to open the Run window.
2. Type `wt` to open Windows Terminal (I've set Nushell as the default shell beforehand).
3. Type `a`.

I get incorrect rendering as follows:

**When starting:**

```nu
┏[ hyoryu][ RAM: 28/63GB][ ~]
┖[ 01:52:45] ♥
```

**After entering `a` (or any other key):**

```nu
┏[ hyoryu][ RAM: 28/63GB][ ~]
┏[ hyoryu][ RAM: 28/63GB][ ~]
┖[ 01:53:21] ♥ a
```

The creator of Oh My Posh cannot reproduce this issue, but I have been able to reproduce it on three different computers(I use GitHub to store the same set of configurations). I have no idea why this happens. Could anyone help me?

Details are available at:

[https://github.com/JanDeDobbeleer/oh-my-posh/discussions/5677](https://github.com/JanDeDobbeleer/oh-my-posh/discussions/5677)

**Comment by fdncred** [comment, marked as answer]
I said this over at the omp repo too. This is what I think is going on, but I could be wrong.

The only time i've seen something like this in nushell is when nushell starts too fast. As I recall, what happens is nushell starts and gets the terminal width, let's say 80x24, in the meantime, your terminal finishes starting and now the terminal width is 100x24, but nushell already knows the size as 80x24, so when you type any character, it realizes it's bigger and redraws the prompt. At least that's my theory. We could introduce some artificial delay in the nushell startup, but I hate to intentionally slow things down.

**Reply by SetsuikiHyoryu** [reply]
@fdncred 

Yes, I'm using Windows Terminal in maximized mode. I tried changing the window size to default, and the issue disappeared. When I tried using a nearly full window size (190 columns * 50 rows), there was also no problem.

Maximizing the window is quite important to me, so I’ll have to put up with this issue. However, I'm grateful to know the possible cause of the problem. Thank you!

**Reply by SetsuikiHyoryu** [reply]
After updating Nushell to 0.99.1, this issue also disappeared when maximized. I'm quite certain it's unrelated to oh-my-posh version 24.1.0, as I tested it after updating.

I looked through the update notes for Nushell versions 0.99.1 and 0.99.0 and didn’t find any directly related fixes. The changes in 0.99.1 are minor, so I guess the issue might have been resolved in 0.99.0.

**Comment by ghost** [comment]
you could check fdncred's theory by running `term size` when the bug occurs, then resize the terminal a tiny bit, rerun it and check if the sizes make sense or weather it defaulted to something on startup.  
if thats the fault you could check weather `wt` allows nushell to resize it via `echo "\e[8;h;wt"` ([source](https://stackoverflow.com/a/5243640)) and if it does you can add `sleep 0.25sec; echo "\e[8;h;wt"` to your config (if the windows gets resized after everything started it should probably refresh the variables).  
another idea would be to just add `sleep 0.25sec` to your config, but i assume that the size gets checked before the config gets loaded.

i frequently have such issues when using a nested terminal emulator ([dvtm](https://www.brain-dump.org/projects/dvtm/), [tmux](https://github.com/tmux/tmux), [screen](https://www.gnu.org/software/screen/), [fterm.nvim](https://github.com/numToStr/FTerm.nvim), [yaft](https://github.com/uobikiemukot/yaft), etc), but those are extremely buggy with any program in any context and get even worse when used via `ssh`.  
therefore i have defaulted to a way more simplistic config and only use the `3x` ANSI-codes (`ansi -l | where code =~ '^\\e\[3\dm$'`).  
i know this isnt ideal, but there are to many things that i would have to patch or completely rewrite to get a fancy prompt working.  
since `starship`, `oh-my-posh`, etc all try to do fancy things i wrote [my shellprompt](https://github.com/Jan9103/nushell_yasp) myself - feel free to take some inspiration (or copy it). it shouldn't be to hard to recreate yours with a similar script / adapted version (unless you have a million `you are using python version x`, `successfully linted all scripts`, etc extensions)

**Reply by SetsuikiHyoryu** [reply]
Thank you for your answer. It is indeed related to the terminal window size, but I don’t really want to sacrifice shell performance for a better appearance, so I’ll probably maintain the status quo. As for the second solution, to be honest, I’m not very familiar with shell, so it’s a bit challenging for me. I might try it at some point. Overall, thank you for your response!
